econml/cate_interpreter.py

```diff
@@ -0,0 +1,586 @@
+        If None then unlimited number of leaf nodes.
+
+    min_impurity_decrease : float, optional, default 0.
+        A node will be split if this split induces a decrease of the impurity
+        greater than or equal to this value.
+
+        The weighted impurity decrease equation is the following::
+
+            N_t / N * (impurity - N_t_R / N_t * right_impurity
+                                - N_t_L / N_t * left_impurity)
+
+        where ``N`` is the total number of samples, ``N_t`` is the number of
+        samples at the current node, ``N_t_L`` is the number of samples in the
+        left child, and ``N_t_R`` is the number of samples in the right child.
+        ``N``, ``N_t``, ``N_t_R`` and ``N_t_L`` all refer to the weighted sum,
+        if ``sample_weight`` is passed.
+    """
+
+    def __init__(self,
+                 include_model_uncertainty=False,
+                 uncertainty_level=.05,
+                 splitter="best",
+                 max_depth=None,
+                 min_samples_split=2,
+                 min_samples_leaf=1,
+                 min_weight_fraction_leaf=0.,
+                 max_features=None,
+                 random_state=None,
+                 max_leaf_nodes=None,
+                 min_impurity_decrease=0.):
+        self.include_uncertainty = include_model_uncertainty
+        self.uncertainty_level = uncertainty_level
+        self.criterion = "mse"
+        self.splitter = splitter
+        self.max_depth = max_depth
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_features = max_features
+        self.random_state = random_state
+        self.max_leaf_nodes = max_leaf_nodes
+        self.min_impurity_decrease = min_impurity_decrease
+
+    def interpret(self, cate_estimator, X):
+        self.tree_model = DecisionTreeRegressor(criterion=self.criterion,
+                                                splitter=self.splitter,
+                                                max_depth=self.max_depth,
+                                                min_samples_split=self.min_samples_split,
+                                                min_samples_leaf=self.min_samples_leaf,
+                                                min_weight_fraction_leaf=self.min_weight_fraction_leaf,
+                                                max_features=self.max_features,
+                                                random_state=self.random_state,
+                                                max_leaf_nodes=self.max_leaf_nodes,
+                                                min_impurity_decrease=self.min_impurity_decrease)
+        y_pred = cate_estimator.const_marginal_effect(X)
+
+        assert (y_pred.ndim == 1) or (y_pred.shape[1] ==
+                                      1), "Only single-dimensional treatment interpretation available!"

```

Comment:
If it's multi dimensional Y, y_pred.shape[1] is the dimension of Y not T. Or we also don't support multi dimensional Y? What should it be if multi Y and also include uncertainty?

Reply:
I've improved the assertion logic and message to make it clear that we only support single-treatment, single-outcome estimates.